### PR TITLE
[TT-2900] Bugfix/SQL total latency + AVG precision loss

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Change the SQL ON CONFLICT AVG formula for `request_time`, `upstream_latency` and `latency` to avoid precision loss. Now it's more simple and we just do a `AVG = (oldTotal + newTotal ) / (oldHits + newHits)`.

Also included `total_latency` to the ON CONFLICT clauses since it was missing.

This apply for `sql_aggregate` and uptime pump
## Related Issue

https://tyktech.atlassian.net/browse/TT-2900
## Motivation and Context

https://tyktech.atlassian.net/browse/TT-2900
## How This Has Been Tested

Added unit tests for sql_aggregate.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: pumps/sql_aggregate_test.go [hunk 1]
```diff
@@ -191,7 +191,7 @@ func TestSQLAggregateWriteDataValues(t *testing.T) {
 		t.Fatal("SQL Pump Aggregate couldn't be initialized with err: ", err)
 	}
 	defer func(table string) {
-		pmp.db.Migrator().DropTable(analytics.AggregateSQLTable)
+		//pmp.db.Migrator().DropTable(analytics.AggregateSQLTable)
```
Comment (vaske) [inline pumps/sql_aggregate_test.go:194]: should we remove entire defer function?

**Review by vaske — APPROVED** [review]
LGTM, I added one minor comment but rest looks ok.

File: pumps/sql_aggregate_test.go (hunk 1 above)
Comment (buger) [reply pumps/sql_aggregate_test.go]: I think it needs be uncommented, since its goal is to clean up after the test.

**Comment by buger** [comment]
@tbuchaillot we really need to add Github Actions CI which runs tests

File: pumps/sql_aggregate_test.go (hunk 1 above)
Comment (tbuchaillot) [reply pumps/sql_aggregate_test.go]: thanks!

**Pull request merged**